xbmc/cores/AudioEngine/AESinkFactory.cpp

```diff
@@ -38,41 +38,53 @@ bool CAESinkFactory::HasSinks()
   return !m_AESinkRegEntry.empty();
 }
 
-void CAESinkFactory::ParseDevice(std::string &device, std::string &driver)
+AESinkDevice CAESinkFactory::ParseDevice(const std::string& device)
 {
-  int pos = device.find_first_of(':');
+  AESinkDevice dev{};
+  dev.name = device;
+
+  size_t pos = dev.name.find_first_of(':');
   bool found = false;
   if (pos > 0)
   {
-    driver = device.substr(0, pos);
+    dev.driver = device.substr(0, pos);
 
     for (const auto& reg : m_AESinkRegEntry)
     {
-      if (!StringUtils::EqualsNoCase(driver, reg.second.sinkName))
+      if (!StringUtils::EqualsNoCase(dev.driver, reg.second.sinkName))
         continue;
 
-      device = device.substr(pos + 1, device.length() - pos - 1);
+      dev.name = dev.name.substr(pos + 1, dev.name.length() - pos - 1);
       found = true;
     }
   }
 
   if (!found)
-    driver.clear();
+    dev.driver.clear();
+
+  pos = dev.name.find_first_of(':');
+  if (pos > 0)
```

Comment:
This (and the same in existing code earlier) is wrong and works by luck.
Try this function with device=WASAPI:default and you will see pos=18446744073709551615 here and the execution enters the block.
suggestion: `if (pos != std::string::npos)`